applications/GeoMechanicsApplication/custom_utilities/dof_utilities.cpp

```diff
@@ -0,0 +1,116 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Anne van de Graaf
+//
+
+#include <algorithm>
+
+#include "dof_utilities.h"
+#include "includes/variables.h"
+
+namespace Kratos
+{
+
+std::vector<std::size_t> ExtractEquationIdsFrom(const std::vector<Dof<double>*>& rDofs)
+{
+    std::vector<std::size_t> result;
+    std::transform(rDofs.begin(), rDofs.end(), std::back_inserter(result),
+                   [](const auto p_dof) { return p_dof->EquationId(); });
+    return result;
+}
+
+Vector ExtractSolutionStepValues(const std::vector<Dof<double>*>& rDofs, int Step)
+{
+    auto result = Vector(rDofs.size());
+    std::transform(rDofs.begin(), rDofs.end(), result.begin(),
+                   [Step](auto p_dof) { return p_dof->GetSolutionStepValue(Step); });
+    return result;
+}
+
+Vector ExtractSolutionStepValuesOfUPwDofs(const std::vector<Dof<double>*>& rDofs, int Step)
+{
+    auto result        = Vector(rDofs.size());
+    auto get_dof_value = [Step](const auto p_dof) -> double {
```

Comment:
I think this could be const (or does the transform take a non-const ref?)
```suggestion
    const auto get_dof_value = [Step](const auto p_dof) -> double {
```

Reply:
The `auto` here deduces the type of the lambda expression. So far, I have never encountered a lambda expression that was assigned to a `const auto` instance. I have browsed through the examples in the C++ Core Guidelines, but they all seem to use just `auto`, so without the `const`. I would suggest to follow those examples, unless you can provide me with an authoritative reference that actually prescribes to use `const auto` for these cases.

Reply:
I was indeed to quick with this comment, I had the same comment for the lambdas in the dof utils, but came to the same conclusion that it does not make too much sense for a lambda expression and removed them again. Totally agree with you that we keep it as is